backend/src/main/java/gov/cdc/usds/simplereport/validators/PatientBulkUploadFileValidator.java

```diff
@@ -37,49 +39,69 @@ public List<FeedbackMessage> validate(InputStream csvStream) {
+    while (valueIterator.hasNext()) {
+      Map<String, String> row;
       try {
         row = getNextRow(valueIterator);
       } catch (CsvProcessingException ex) {
-        errors.add(
+        var feedback =
             new FeedbackMessage(
                 CsvValidatorUtils.REPORT_SCOPE,
                 "File has the incorrect number of columns or empty rows. Please make sure all columns match the data template, and delete any empty rows.",
-                new int[] {ex.getLineNumber()}));
-        return errors;
+                List.of(ex.getLineNumber()));
+        mapOfErrors.put(feedback.getMessage(), feedback);
+        break;
       }
+      var currentRowErrors = new ArrayList<FeedbackMessage>();
 
       PatientUploadRow extractedData = new PatientUploadRow(row);
 
-      errors.addAll(extractedData.validateHeaders());
+      currentRowErrors.addAll(extractedData.validateHeaders());
 
       // validate individual values
 
       // demographics
-      errors.addAll(validateDate(extractedData.dateOfBirth));
-      errors.addAll(validateRace(extractedData.race));
-      errors.addAll(validateBiologicalSex(extractedData.biologicalSex));
-      errors.addAll(validateEthnicity(extractedData.ethnicity));
+      currentRowErrors.addAll(validateDate(extractedData.dateOfBirth));
+      currentRowErrors.addAll(validateRace(extractedData.race));
+      currentRowErrors.addAll(validateBiologicalSex(extractedData.biologicalSex));
+      currentRowErrors.addAll(validateEthnicity(extractedData.ethnicity));
 
       // housing, work, and role
-      errors.addAll(validateYesNoAnswer(extractedData.residentCongregateSetting));
-      errors.addAll(validateYesNoAnswer(extractedData.employedInHealthcare));
-      errors.addAll(validateRole(extractedData.role));
+      currentRowErrors.addAll(validateYesNoAnswer(extractedData.residentCongregateSetting));
+      currentRowErrors.addAll(validateYesNoAnswer(extractedData.employedInHealthcare));
+      currentRowErrors.addAll(validateRole(extractedData.role));
 
       // address
-      errors.addAll(validateState(extractedData.state));
-      errors.addAll(validateZipCode(extractedData.zipCode));
-      errors.addAll(validateCountry(extractedData.country));
+      currentRowErrors.addAll(validateState(extractedData.state));
+      currentRowErrors.addAll(validateZipCode(extractedData.zipCode));
+      currentRowErrors.addAll(validateCountry(extractedData.country));
 
       // contact info
-      errors.addAll(validatePhoneNumber(extractedData.phoneNumber));
-      errors.addAll(validatePhoneNumberType(extractedData.phoneNumberType));
-      errors.addAll(validateEmail(extractedData.email));
+      currentRowErrors.addAll(validatePhoneNumber(extractedData.phoneNumber));
+      currentRowErrors.addAll(validatePhoneNumberType(extractedData.phoneNumberType));
+      currentRowErrors.addAll(validateEmail(extractedData.email));
+
+      final var finalCurrentRow = currentRow;
```

Comment:
Why do we need to redefine `currentRow` in a new var? I don't see it being modified before incrementing 🤔

Reply:
This is a recommendation by intellij to not use non final variables in lambdas. [Read more here](https://stackoverflow.com/questions/34865383/variable-used-in-lambda-expression-should-be-final-or-effectively-final)
![image](https://user-images.githubusercontent.com/10108172/202535167-c27d2e9e-f402-40c4-9592-42997cb45403.png)